java-frontend/src/main/java/org/sonar/java/se/checks/CustomUnclosedResourcesCheck.java

```diff
@@ -45,13 +46,22 @@
 @RuleTemplate
 public class CustomUnclosedResourcesCheck extends SECheck {
 
-  static class ResourceStatus implements ObjectConstraint.Status {
+  public class CustomResourceConstraint implements Constraint {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this a "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694)